onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -556,8 +564,33 @@ void RegisterExecutionProviders(InferenceSession* sess, const std::vector<std::s
 #endif
     } else if (type == kOpenVINOExecutionProvider) {
 #ifdef USE_OPENVINO
-      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_OpenVINO(openvino_device.c_str()));
-      openvino_device.clear();
+      auto it = provider_options_map.find(type);
+      if(it != provider_options_map.end()) {
+        for(auto option : it->second) {
+          if(option.first == "device_type") openvino_device_type = option.second;
+          else if (option.first == "enable_vpu_fast_compile") {
+            if(option.second == "True") {
+              enable_vpu_fast_compile = true;
+            } else if (option.second == "False") {
+              enable_vpu_fast_compile = false;
+            } else {
+              ORT_THROW("Invalid value passed for enable_vpu_fast_compile: ", option.second);
+            }
+
+          }
+          else if (option.first == "device_id")  openvino_device_id = option.second;
+          else {
+            ORT_THROW("Invalid OpenVINO EP option: ", option.first);
+          }
+        }
+      }
+      RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_OpenVINO(openvino_device_type.c_str(),
+                                                                                            enable_vpu_fast_compile,
+                                                                                            openvino_device_id.c_str()));
+      // Reset config settings to avoid it being accidentally passed on to the next session
+      openvino_device_type.clear();
```

Comment:
should make openvino_device_type, id, enable_vpu_fast_compile local. then you don't need to do this.